andela/travel_tool_back
# #162006041 Fix create multi city request

#### What does this PR do?
Allows a user to create any type of request flawlessly

#### Description of Task to be completed?
- Refactor the endpoint for creating a request so that users can make oneWay, multi and return trips.
- Refactor tests for the endpoint

#### How should this be manually tested?
- Clone the repo `git clone https://github.com/andela/travel_tool_back.git`
- Run `cd travel_tool_back` to switch to the project directory
- Run `make start` to install dependencies and start the app
- Login to the travela app. After you are logged in, visit the developer console in your browser and open the `Application` tab, you should see your `jwt-token` under cookies.
- Copy the token and add it to the `authorization` header in `postman`
- In `postman`, make a `POST` request to the endpoint `http://localhost:5000/api/v1/requests/` with all of following fields as shown in the screenshot. You should get a 201 response with the created request.

#### Any background context you want to provide?
This tasks works  with notification. To ensure the manager is notified, first create the manager in the Users table.
Also note that a oneWay trip cannot have a return date, a return trip must have a departure date and a return date and  a multi trip must have more than one return trip

#### What are the relevant pivotal tracker stories?
[162006041 ](https://www.pivotaltracker.com/story/show/162006041 )

#### Screenshots (if appropriate)

#### Questions:
None
